File: Domain/Classes/Book.cs
using Domain.Enums;
using Domain.Interfaces;
using Domain.ValueObjects;

namespace Domain.Classes;

public sealed class Book : IEntity, IUserStats, IAdded
{
  public int Id { get; set; }
  public string Title { get; set; } = null!;
  public string? Description { get; set; }
  public short? Chapters { get; set; }
  public short? Pages { get; set; }
  public int? Words { get; set; }
  public DateOnly? Published { get; set; }
  public Image Cover { get; set; } = null!;
  public decimal Score { get; set; }
  public DateOnly Added { get; set; }
  public int Popularity { get; set; }
  public int Favorites { get; set; }
  public int Members { get; set; }
  public int Rank { get; set; }

  public Genre[] Genres { get; set; } = Array.Empty<Genre>();
  public Theme[] Themes { get; set; } = Array.Empty<Theme>();
}

public sealed class CreateBook
{
  public int Id { get; set; }
  public short? Chapters { get; set; }
  public short? Pages { get; set; }
  public int? Words { get; set; }
  public DateOnly? Published { get; set; }
  public CreateImage? Cover { get; set; } = null!;

  public Genre[]? Genres { get; set; } = Array.Empty<Genre>();
  public Theme[]? Themes { get; set; } = Array.Empty<Theme>();

  public Dictionary<Language, TitleDescription>? Translations { get; set; } = null!;
}

File: domain/entities/book/book.rs
use chrono::NaiveDate;

use crate::entities::franchise::franchise::Franchise;
use crate::entities::image::image::Image;

#[derive(Debug)]
#[cfg_attr(feature = "serde", derive(serde::Serialize))]
#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
pub struct Book {
  pub id: i32,
  pub title: String,
  pub description: Option<String>,
  pub chapters: Option<i16>,
  pub pages: Option<i16>,
  pub words: Option<i32>,
  #[cfg_attr(feature = "utoipa", schema(schema_with = crate::utoipa::utoipa::naive_date_optional))]
  pub published: Option<NaiveDate>,
  pub cover: Image,
  pub score: f32,
  pub rank: i32,
  pub popularity: i32,
  pub favorites: i32,
  pub members: i32,
  #[cfg_attr(feature = "utoipa", schema(schema_with = crate::utoipa::utoipa::naive_date))]
  pub added: NaiveDate,
  pub franchise: Option<Franchise>,
}
